File: src/main/java/org/intellij/sdk/language/psi/SimpleElementFactory.java
// Copyright 2000-2022 JetBrains s.r.o. and other contributors. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.

package org.intellij.sdk.language.psi;

import com.intellij.openapi.project.Project;
import com.intellij.psi.PsiElement;
import com.intellij.psi.PsiFileFactory;
import org.intellij.sdk.language.SimpleFileType;
import org.intellij.sdk.language.psi.impl.SimpleTypeDeclarationImpl;

public class SimpleElementFactory {
    public static SimpleFile createFile(Project project, String text) {
        return (SimpleFile) PsiFileFactory.getInstance(project).createFileFromText("d.simple", SimpleFileType.INSTANCE, text);
    }

    public static PsiElement createIdentifier(Project project, String name) {
        final SimpleFile file = createFile(project, "type " + name + "{}");
        return ((SimpleTypeDeclarationImpl) file.getFirstChild()).getIdentifier();
    }
}

File: src/main/kotlin/com/example/sampleplugin/psi/SimpleElementFactory.kt
package com.example.sampleplugin.psi

import com.example.sampleplugin.SimpleFileType
import com.intellij.openapi.project.Project
import com.intellij.psi.PsiFileFactory
import com.intellij.psi.util.childrenOfType

object SimpleElementFactory {
    @JvmStatic
    fun createIdentifier(project: Project?, name: String): SimpleNamedElement {
        val file = createFile(project, "type $name {}")
        return file.firstChild.childrenOfType<SimpleNamedElement>().first()
    }

    fun createFile(project: Project?, text: String?): SimpleFile {
        return text?.let {
            PsiFileFactory.getInstance(project).createFileFromText("dummy.simple", SimpleFileType.INSTANCE, it)
        } as SimpleFile
    }
}
